### PR TITLE
Chore: Use 'folderPath' instead of 'asset' during publishing

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Changelog Description
Rename `"asset"` key used during publishing to `"folderPath"`. This is first of many PRs changing asset naming to folder.

## Additional info
I think resolve, hiero and traypublisher might not be converted correctly. Please @jakubjezek001 test them and if you find and issue try to fix it.

## Testing notes:
Ideal test scenario:
1. Open a host with workfile that already did have created instances.
2. Use all possible creators.
3. Publish all the instances.
4. Everything is working and shiny.
- [x] 3dsMax
- [x] AfterEffects
- [x] Blender
- [x] Fusion
- [x] Hiero
- [x] Houdini
- [x] Maya
- [x] Nuke
- [x] Photoshop
- [x] Resolve
- [x] SubstancePainter
- [x] TrayPublisher
- [x] Unreal
- [ ] ~Celaction~
- [ ] ~Harmony~
- [ ] ~TVpaint~

I do expect that hosts using Publisher tool should not have any issues. Possible issues are with hosts using pyblish pype, automated publishin login and editorial.

### CODE REVIEW

**Comment by iLLiCiTiT** [comment]
> Should this line be asset_name=context.data["folderPath"], ?

Yes, well spotted (I missed this because of other changes).

**Comment by moonyuet** [comment]
collect_workfile.py and collect_render.py in 3dsMax hosts have both `context.data["asset"]` and `instance.data["asset"]` assigned. 
And also collect_anatomy_instance_data.py in plugins folder is still using `instance.data["asset"] `

**Comment by iLLiCiTiT** [comment]
I had collapsed hosts when I've beek picking the lines to commit. Damn, should be fine now.

File: client/ayon_core/plugins/publish/collect_anatomy_instance_data.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def fill_missing_asset_docs(self, context, project_name):
         instances_with_missing_asset_doc = collections.defaultdict(list)
         for instance in context:
             instance_asset_doc = instance.data.get("assetEntity")
-            _asset_name = instance.data["asset"]
+            _asset_name = instance.data["folderPath"]
```
Comment (BigRoy) [inline client/ayon_core/plugins/publish/collect_anatomy_instance_data.py:71]: Should we also be refactoring the variable names here so it's clear that these are now folder paths? There are quite a few still referring to `asset` and `asset_name`.

**Comment by moonyuet** [comment]
Tested with 3dsmax, Looks okay now with all families.

File: client/ayon_core/plugins/publish/collect_anatomy_instance_data.py (hunk 1 above)
Comment (iLLiCiTiT) [reply client/ayon_core/plugins/publish/collect_anatomy_instance_data.py:71]: Yes, that's what I have prepared. But I wanted to keep this PR as simple as possible.

**Comment by iLLiCiTiT** [comment]
> Or my plugin is not updated

Looks like not updated if it looks for `AVALON_` environments.

**Review by simonebarbieri — APPROVED** [review]
I've tested Blender and Unreal and they seem to be working fine!

**Review by MustafaJafar — COMMENTED** [review]
Hello, 
I tested this PR from Houdini. ~~everything works as usual.~~ not all product types works as usual. still testing.
small question: 
In different Houdini pyblish plugins I can find things like: 
- `context["asset"]` 
- `dynamic_data["asset"]` 
- `instance.data["asset"]`
- `dependency.data["asset"]` 
Should these be included in this PR ? 

---
Also, even publishing programmatically using `pyblish.util` works. 
For reference I used this code: 
```python
import hou
import logging
import pyblish.util
from ayon_core.pipeline import registered_host
from ayon_core.pipeline.create import CreateContext

log = logging.getLogger(__name__)

all_input_nodes = [
    hou.node('/out/reviewMain'),
    hou.node('/out/S_BigRoboMain')
]

comment = "publish from code"

all_input_subsets = [n.evalParm("subset") for n in all_input_nodes]

host = registered_host()
context = CreateContext(host, reset=True)

for instance in context.instances:
    if instance.get("subset") not in all_input_subsets:
        instance["active"] = False
        instance.data["publish"] = False
        continue

    # make sure the instance is active
    instance["active"] = True

context.save_changes()

pyblish_context = pyblish.api.Context()
pyblish_context.data["create_context"] = context
pyblish_context.data["comment"] = comment

pyblish_plugins = context.publish_plugins

error_format = "Failed {plugin.__name__}: {error} -- {error.traceback}"

for result in pyblish.util.publish_iter(pyblish_context, pyblish_plugins):
    for record in result["records"]:
        log.debug("{}: {}".format(result["plugin"].label, record.msg))

    # Exit as soon as any error occurs.
    if result["error"]:
        error_message = error_format.format(**result)
        log.debug(error_message)
```

**Comment by MustafaJafar** [comment]
Any ideas why this occurs in Houdini when publishing `"imagesequence"`
```python
CRITICAL: Error when registering
Traceback (most recent call last):
  File "E:\Ynput\ayon-core\client\ayon_core\plugins\publish\integrate.py", line 182, in process
    self.register(instance, file_transactions, filtered_repres)
  File "E:\Ynput\ayon-core\client\ayon_core\plugins\publish\integrate.py", line 388, in register
    op_session.commit()
  File "E:\Ynput\ayon-core\client\ayon_core\client\operations.py", line 747, in commit
    raise FailedOperations((
ayon_core.client.operations.FailedOperations: Operation "079ed864-235f-48cd-8f8d-bc05c5d1e322" failed with data:
{
    "id": "079ed864-235f-48cd-8f8d-bc05c5d1e322",
    "type": "create",
    "entityType": "representation",
    "entityId": "f412f2f3cf5c11eebc722f62e1d11282",
    "data": {
        "name": "",
        "versionId": "f411c1c2cf5c11ee806b2f62e1d11282",
        "id": "f412f2f3cf5c11eebc722f62e1d11282",
        "files": [
            {
                "path": "{root[work]}/Robo/Assets/Character/BigRobo/publish/imagesequence/imagesequenceMain/v012/robo_BigRobo_imagesequenceMain_v012.",
                "size": 25294,
                "hash": "robo_BigRobo_imagesequenceMain_v012,|1708370858,50869|25294",
                "id": "f430fe6fcf5c11eeb3442f62e1d11282",
                "hash_type": "op3",
                "name": "robo_BigRobo_imagesequenceMain_v012."
            }
        ],
        "attrib": {
            "path": "\\\\storage\\work\\ayon_projects\\Robo\\Assets\\Character\\BigRobo\\publish\\imagesequence\\imagesequenceMain\\v012\\robo_BigRobo_imagesequenceMain_v012.",
            "template": "{root[work]}\\{project[name]}\\{hierarchy}\\{folder[name]}\\publish\\{product[type]}\\{product[name]}\\v{version:0>3}\\{project[code]}_{folder[name]}_{product[name]}_v{version:0>3}<_{output}><.{frame:0>4}><_{udim}>.{ext}"
        },
        "data": {
            "context": {
                "root": {
                    "work": "\\\\storage\\work\\ayon_projects"
                },
                "project": {
                    "name": "Robo",
                    "code": "robo"
                },
                "hierarchy": "Assets/Character",
                "folder": {
                    "name": "BigRobo"
                },
                "version": 12,
                "ext": "",
                "asset": "BigRobo",
                "task": {
                    "name": "cfx",
                    "type": "CFX",
                    "short": "cfx"
                },
                "representation": "",
                "username": "admin",
                "user": "admin",
                "product": {
                    "type": "imagesequence",
                    "name": "imagesequenceMain"
                }
            }
        }
    }
}
Error: unknown.
```

**Comment by tokejepsen** [comment]
Not sure whether this is outside of the scope of this PR but found this:
```
Traceback (most recent call last):
  File "C:\Users\tokejepsen\AppData\Local\Ynput\AYON\dependency_packages\ayon_2402141620_windows.zip\dependencies\pyblish\plugin.py", line 527, in __explicit_process
    runner(*args)
  File "C:\Users\tokejepsen\AppData\Local\Ynput\AYON\addons\ftrack_1.0.5-dev.1\ayon_ftrack\plugins\publish\collect_ftrack_api.py", line 36, in process
    folder_path = context.data["asset"]
KeyError: 'asset'
```

**Comment by tokejepsen** [comment]
Defo outside the scope of this PR, but not sure where to post:
```
2024-02-20 08:28:15:  0: PYTHON: Traceback (most recent call last):
2024-02-20 08:28:15:  0: PYTHON:   File "C:\ProgramData\Thinkbox\Deadline10\workers\DESKTOP-969549J\plugins\65d461c2b0095d07d10f1ca0\GlobalJobPreLoad.py", line 488, in inject_ayon_environment
2024-02-20 08:28:15:  0: PYTHON:     raise RuntimeError((
2024-02-20 08:28:15:  0: PYTHON: RuntimeError: Missing required env vars: AVALON_PROJECT, AVALON_ASSET, AVALON_TASK, AVALON_APP_NAME
2024-02-20 08:28:15:  0: PYTHON: !!! Injection failed.
```
Trying to test render on Deadline.

**Comment by BigRoy** [comment]
> Defo outside the scope of this PR, but not sure where to post:
> 
> ```
> 2024-02-20 08:28:15:  0: PYTHON: Traceback (most recent call last):
> 2024-02-20 08:28:15:  0: PYTHON:   File "C:\ProgramData\Thinkbox\Deadline10\workers\DESKTOP-969549J\plugins\65d461c2b0095d07d10f1ca0\GlobalJobPreLoad.py", line 488, in inject_ayon_environment
> 2024-02-20 08:28:15:  0: PYTHON:     raise RuntimeError((
> 2024-02-20 08:28:15:  0: PYTHON: RuntimeError: Missing required env vars: AVALON_PROJECT, AVALON_ASSET, AVALON_TASK, AVALON_APP_NAME
> 2024-02-20 08:28:15:  0: PYTHON: !!! Injection failed.
> ```
> 
> Trying to test render on Deadline.

@tokejepsen have you updated your Deadline's `GlobalJobPreLoad.py` file?

**Comment by iLLiCiTiT** [comment]
> Not sure whether this is outside of the scope of this PR but found this:

Please disable ftrack. This PR will break backwards compatibility with already splitted addons (ftrack, shotgrid, slack, etc.).

> Trying to test render on Deadline.

As BigRoy commented, please update deadline plugin.

**Review by tokejepsen — APPROVED** [review]
> @tokejepsen have you updated your Deadline's GlobalJobPreLoad.py file?

Sorry, yeah I did update it, but the cached files on the worker had the old version. So annoying the caching is not reliable.

Tested in Maya 2023 successfully.

**Comment by tokejepsen** [comment]
Nuke 15 tested successfully.

**Comment by MustafaJafar** [comment]
>  `instance.data["asset"]` - These are wrong. Found 5 of these, and fixed them. If you find more, then report.


I found `inst.data["asset"]`  here
https://github.com/ynput/ayon-core/blob/enhancement/OP-8208_Use-folderPath-during-publishing/client/ayon_core/hosts/houdini/plugins/publish/collect_usd_bootstrap.py

**Comment by moonyuet** [comment]
Tested in Substance Painter, works smoothly

**Review by jakubjezek001 — CHANGES_REQUESTED** [review]
image

Traypublisher Editorial  Simple is giving me following results

[publish-report-240220-15-41.json](https://github.com/ynput/ayon-core/files/14345887/publish-report-240220-15-41.json)

**Review by MustafaJafar — APPROVED** [review]
Tested in Houdini. Works as usual. 
Thanks.

**Review by 64qam — APPROVED** [review]
all good in AE and PS

**Review by jakubjezek001 — APPROVED** [review]

**Pull request merged**